### ISSUE TITLE
Error number 0042 gives bad error string

### ISSUE THREAD

**Issue by prbsparx** [rtopken/CalCheck#5]
Error 0042 - "ERROR: The   organizer of this meeting is potentially incorrect. See KB 2563324:   https://support.microsoft.com/en-us/kb/2563324"

This error message references a KB article that doesn't exist anymore. Please update to show the correct KB article information.

**Comment by stephenegriffin** [comment]
https://web.archive.org/web/20190115122408/https://support.microsoft.com/en-us/help/2563324/current-issues-with-microsoft-exchange-activesync-and-third-party-devi

**Comment by prbsparx** [comment]
I saw the web archive, but the KB was deleted from MS site for a reason. It'd be good to have clarity on what action the user should take to correct that error message. I don't see any item in the old KB that clearly describes how to fix these ones.

**Comment by rtopken** [comment]
You can look at this article for more info as well. I will post it in the wiki.
[https://support.microsoft.com/en-us/help/2678030/information-about-the-calendar-checking-tool-for-outlook-calcheck](https://support.microsoft.com/en-us/help/2678030/information-about-the-calendar-checking-tool-for-outlook-calcheck)
In general - if the organizer of an Outlook Meeting changed - that is a problem. And the item should be deleted/removed and recreated.
That said - if you recently migrated to a new Exchange organization (like to the Microsoft O365 service) then your users will have new x.500 addresses reflecting this. 
The test in CalCheck here is looking to see:

1. Is this user the Organizer of the meeting? There are props that tell if this user is supposed to be the Organizer.
2. If 1 is true, then does the x.500 address set for the organizer on the item match the x.500 address of this user's mailbox?

When a migration occurs and the x.500 address changes, then this can give a false error. The tool tries to get all the proxy addresses for this mailbox and will test against them and one of them should be the x.500 address of the previous Exchange Org, but sometimes that does not work and so you end up getting a false error reported.

So you can check the Organizer on the item and see if the address is really correct or not. If it's not then it's generally best to delete and recreate the item.

**Comment by prbsparx** [comment]
Hey @rtopken - I looked at the Microsoft KB initially, that's how I got the error number.

I'll give you an actual example we ran into when running CalCheck.

$user uses Outlook for Mac. He scheduled a meeting with me so we could setup his email in Outlook for Windows. The meeting was originally set for 3-4:30 pm yesterday. He moved it to 3:30 - 5pm because of a conflict.

When we ran CalCheck, the report included that meeting with this error. ("ERROR: The organizer of this meeting is potentially incorrect.")

The organizer looks right both in Outlook for Mac and Outlook for Windows. Do we then have to run MFCMAPI against the item directly to investigate it further to determine why it's reporting that or is there an easier way?

**Comment by rtopken** [comment]
Yeah - MFCMAPI would be the way to inspect the MAPI props on the item. The props to look for are
PR_SENT_REPRESENTING_EMAIL_ADDRESS
pidlidAppointmentStateFlags

PR_SENT_REPRESENTING_EMAIL_ADDRESS - should match the user/mailbox that is the organizer.
pidlidAppointmentStateFlags - should be “1” in the Organizer’s mailbox. Should be “3” in an attendees mailbox.

And again - if the address in PR_SENT_REPRESENTING_EMAIL_ADDRESS does not match the x.500 address of the Organizer’s mailbox - the tool will flag it for an error. But if the address there is an address for the Organizer - then it is a false error and can be safely ignored.

-Randy

**Comment by prbsparx** [comment]
PR_SENT_REPRESENTING_EMAIL_ADDRESS looks like an OU description of the user.
/O=COMPANY/OU=LOCATION/CN=CONTAINER/CN=LASTNAME, FIRSTNAME

How do I verify that this matches the user/mailbox that is the organizer?

**Comment by rtopken** [comment]
The CalCheck.log file from running CalCheck will show you:

Opening mailbox: <display name>

DN: /o=<Org>/ou=<Site>/cn=Recipients/cn=<User>
